app/web_framework/cluster_base_handler.py

```diff
@@ -86,6 +89,10 @@ def prepare(self):
         """
         self._check_expected_session_id()
 
+        accept_header = self.request.headers.get('Accept')
+        uri = self.request.uri
+        self.api_version = APIVersionHandler.resolve_version(accept_header, uri)
```

Comment:
That's a good idea -- do you think we should just include a header with the API version or rather try to mimic what GitHub does here? So like

```
HTTP/1.1 200 OK
Api-Version: 2
...
```

or

```
HTTP/1.1 200 OK
Clusterrunner-Media-Type: clusterrunner.v2; format=json
...
```

I personally like the first idea of just having an `Api-Version` header since the version being served is the only real unique information we're providing. Also we already do something like this with the `Session-Id` response header.